### PR TITLE
Add GitLab support

### LINKED ISSUE #657: Gitlab support
Right now, install tries to use `codeload.github.com` to fetch any non-local package. This makes it impossible for fisher to install plugins from GitLab or any other hosts. I found #337, which indicates that gitlab was supported at one point, can support for gitlab (and potentially other hosts)  be restored?

### PR DESCRIPTION
**Pull request by ainmosni**
Add the possibility to add plugins from gitlab. Resolves #657.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -31,6 +31,12 @@ Install plugins using the `install` command followed by the path to the reposito
 fisher install ilancosman/tide
 ```
 
+To install a plugin from GitLab, prefix `gitlab.com/`.
+
+```console
+fisher install gitlab.com/ainmosni/fish-aws
+```
+
```
Comment (jorgebucaran) [inline README.md:39]: @ainmosni If you can turn this into one sentence and put it after a `>` (we don't need the example), then I'd be happy to include this in the README.

Comment (ainmosni) [reply README.md]: Done.

Comment (jorgebucaran) [reply README.md]: Thank you. I'll think a bit more about it.

File: functions/fisher.fish [hunk 2]
```diff
@@ -84,6 +84,12 @@ function fisher --argument-names cmd --description "A plugin manager for Fish"
                         set name (string split \@ $plugin) || set name[2] HEAD
                         set url https://codeload.github.com/\$name[1]/tar.gz/\$name[2]
 
+                        if string match -q 'gitlab.com/*' \$name[1]
+                            set bare (string replace 'gitlab.com/' '' \$name[1])
+                            set repo (string split '/' \$bare)
+                            set url https://gitlab.com/{\$bare}/-/archive/{\$name[2]}/{\$repo[2]}-{\$name[2]}.tar.gz
```
[the remark this answers is not in the document]
Comment (jorgebucaran) [reply functions/fisher.fish:90]: Interesting. I wonder what other edge cases we may be missing for Gitlab.

**Comment by tennox** [comment]
I'd really like to have this, did your thinking bring something up? :yum:

File: functions/fisher.fish (hunk 2 above)
[the remark this answers is not in the document]
Comment (jorgebucaran) [reply functions/fisher.fish:90]: Thank you for catching this bug. 💯

**Comment by jorgebucaran** [comment]
Available in 4.3.5. Thank you, @ainmosni. 🙌

**Pull request merged**